.github/scripts/common.py

```diff
@@ -173,22 +173,9 @@ def upload(file_to_upload, host, dest_path):
     else:
         port = "22"
 
-    command = [
-        "ssh",
-        "-p",
-        port,
-        "-i",
-        _environ.get("SSH_KEY"),
-        "-o",
-        "StrictHostKeyChecking=no",
-        host,
-        "mkdir -p {}".format(dest_path),
-    ]
-    print_message("Creating dest path {}", dest_path)
-    subprocess.check_call(command)
-
     command = [
         "scp",
+        "-r",
```

Comment:
Have you tested that ?
It seems that `-r` allow to copy recursively directory.
But here, we are copying a file into a (non existent) directory.

I'm affraid, it copies the file and rename it to the directory name (instead of creating) the directory.

Reply:
You're right, it creates appropriate folders if you pass it a directory as source. Let me come back to you

Reply:
Here's another that's slightly different.
I restored the mkdir step but instead of invoking an arbitrary (`mkdir -p`) command
on the remote shell via ssh, it sends `mkdir` SFTP commands.

Unfortunately, sftp isn't smart enough to automagically create the missing remote
folders so we can't just swap it all with a single SFTP command.

This has been tested and I checked that the running images (bionic, f35, focal) do
bundle the sftp binary.

`subprocess.run()` is 3.5+ which is fine given bionic has the oldest as 3.6

Let me know is this is OK with you and I'll squash it all.